src/main/java/core/basesyntax/dao/impl/SmileDaoImpl.java

```diff
@@ -12,16 +15,41 @@ public SmileDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Smile create(Smile entity) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(entity);
+            transaction.commit();
+            return entity;
+        } catch (RuntimeException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't create smile: " + entity, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public Smile get(Long id) {
-        return null;
+        try (Session session = factory.openSession()) {
+            return session.get(Smile.class, id);
+        } catch (Exception e) {
+            throw new RuntimeException("Can't get smile from DB with this id: " + id, e);
+        }
     }
 
     @Override
     public List<Smile> getAll() {
-        return null;
+        try (Session session = factory.openSession()) {
+            Query<Smile> getAllCommentQuery = session
+                    .createQuery("select c from smiles c", Smile.class);
+            return getAllCommentQuery.getResultList();
```

Comment:
I would simplify to this expression:
return session.createQuery("FROM Smile",  Smile.class).getResultList();

Reply:
if i write only "From smile" its dont work, I try a lot of times